### PR TITLE
Update Travis to Xcode 9.2 #trivial

### PR DESCRIPTION
**Pull request by rnystrom**
Issue fixed: #1009

### CODE REVIEW

**Comment by jessesquires** [comment]
@rnystrom if any of these fail, i've figured out the magic numbers here:
https://github.com/jessesquires/JSQCoreDataKit/blob/develop/.travis.yml

File: .travis.yml [hunk 1]
```diff
@@ -15,33 +15,31 @@ env:
...
-   - DESTINATION="OS=9.3,name=iPhone 6s" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="NO" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="NO" POD_LINT="NO"
-
-   - DESTINATION="OS=10.0,name=iPhone 6s"  SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="YES" RUN_UI_TESTS="NO" POD_LINT="NO"
-   - DESTINATION="OS=10.1,name=iPhone 6s"  SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="NO" POD_LINT="NO"
-   - DESTINATION="OS=10.2,name=iPad Air 2" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="NO" POD_LINT="NO"
-   - DESTINATION="OS=10.3.1,name=iPad Air 2" SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="NO"  RUN_UI_TESTS="NO" POD_LINT="NO"
-
-   - DESTINATION="OS=9.0,name=Apple TV 1080p"  SDK="$TVOS_SDK" SCHEME="$TVOS_SCHEME" RUN_TESTS="NO"  BUILD_EXAMPLE="YES" RUN_UI_TESTS="NO" POD_LINT="NO"
-   - DESTINATION="OS=10.0,name=Apple TV 1080p" SDK="$TVOS_SDK" SCHEME="$TVOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="YES" RUN_UI_TESTS="NO" POD_LINT="NO"
+   - DESTINATION="OS=9.0,name=iPhone 6 Plus"  SDK="$IOS_SDK" SCHEME="$IOS_SCHEME" RUN_TESTS="YES" BUILD_EXAMPLE="YES" POD_LINT="YES"
```
Comment (jessesquires) [inline .travis.yml]: shouldn't have copied that other `.travis.yml` wholesale 😅 

- missing `RUN_UI_TESTS`
- missing `CHECK_MARKDOWN` (on first entry)
- missing iOS 8 tests (my library dropped iOS 8 support)

**Comment by jessesquires** [comment]
@rnystrom the full matrix is getting pretty big (once we re-add iOS 8), but i guess it is "fool proof".

we'd probably be ok just using the earliest major version though. (8.1, 9.0, 10.0 only, etc) what do you think?

File: .travis.yml (hunk 1 above)
Comment (rnystrom) [reply .travis.yml]: loool dammit, ok updating

**Comment by jessesquires** [comment]
Dead links found:
https://travis-ci.org/Instagram/IGListKit/jobs/318289838

`https://github.com/Instagram/IGListKit/blob/master/Source/IGListSectionType.h#L43-L54`

**Comment by jessesquires** [comment]
Lots of build warnings and legit test failures here:

https://travis-ci.org/Instagram/IGListKit/jobs/318289839

@rnystrom does this need to be rebased?

**Comment by jessesquires** [comment]
pushed a fix for one of the errors

**Comment by jessesquires** [comment]
Other fixes at #1038.

We should merge that, then rebased this PR.

**Comment by jessesquires** [comment]
dead link issue fixed at #1041

we should also merge that, then rebase this. then i think we'll be green 💚

**Comment by rnystrom** [comment]
some day this will work again

**Comment by rnystrom** [comment]
UI tests have been broken since [June](https://github.com/Instagram/IGListKit/commit/33f88b43e8ede36e6be5efeddf54d4000c05bf9a#diff-732740f542a18f685730e4331fb6bd6d) 😭

_I **will** fix this PR!_

**Comment by jessesquires** [comment]
@rnystrom 👏 

If most things are passing, I think we should go ahead and merge, then followup

**Comment by jessesquires** [comment]
could also set `RUN_UI_TESTS` to `NO` for this PR. then at least other PRs can have proper CI 😄 

then we can followup

**Comment by rnystrom** [comment]
@jessesquires true, if CI comes back ❌  I'll just kill the UI tests and fix them later

**Comment by rnystrom** [comment]
Travis is the slowest thing ever

**Comment by jessesquires** [comment]
@rnystrom - i just pushed a commit to this branch (37b7c03f7bdd13531a6ff12595431a8164ef221a) and merged master. this should fix everything.

- Update to CP 1.4.0
- Add a new `scripts/setup.sh` to do all initial setup after checkout
- Update gems
- Update README, noting to run `setup.sh` after checkout

# DO NOT FORCE PUSH TO THIS BRANCH, PULL FIRST 😄

**Comment by jessesquires** [comment]
travis must be experiencing some issues with their macOS machines.

my other builds on linux have been fine lately.

**Comment by jessesquires** [comment]
canceled the 'branch push' build -- only the pr build matters

**Pull request closed without merging**